### ISSUE TITLE
Bug with compressing files.

### ISSUE THREAD

**Issue by gagan0723** [ovhemert/gatsby-plugin-brotli#9]
I tested the latest version of this plugin with [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default.git) and it doesn't work when only `js` option is passed. It also breaks when `html` is passed as option. Below are the details:

1. It doesn't compress files if only `js` option is passed.

Config:
```js
{
  resolve: "gatsby-plugin-brotli",
    options: {
      extensions: ["js"],
      path: "brotli"
  }
}
```
Plugin code with `console.log()`:
```diff
async function onPostBuild (args, pluginOptions) {
+  console.log("PluginOptions:", pluginOptions)
  const options = { ...defaultOptions, ...pluginOptions }
+  console.log("AllOptions:", options)
  const fileBasePath = path.join(process.cwd(), 'public')
  const pattern = `**/*.{${options.extensions.join(',')}}`
+  console.log("Pattern:", pattern)

  const files = await globAsync(pattern, { cwd: fileBasePath, ignore: '**/*.br' })
+  console.log(files);
  const compress = files.map(file => {
    return compressFile(file, pluginOptions)
  })
  return Promise.all(compress)
}
```
Output on Terminal:
```bash
success Building static HTML for pages - 0.489 s — 4/4 26.21 pages/second
PluginOptions: { plugins: [], extensions: [ 'js' ], path: 'brotli' }
AllOptions: { extensions: [ 'js' ], path: 'brotli', plugins: [] }
Pattern: **/*.{js}
[]
info Done building in 9.861 sec
```
2. It compresses files if both `js` and `css` option are passed.
Config:
```js
{
  resolve: "gatsby-plugin-brotli",
    options: {
      extensions: ["js", "css"],
      path: "brotli"
  }
}
```
Plugin code with `console.log()`: Same as above.
Output on Terminal:
```bash
success Building static HTML for pages - 0.376 s — 4/4 26.97 pages/second
PluginOptions: { plugins: [], extensions: [ 'js', 'css' ], path: 'brotli' }
AllOptions: { extensions: [ 'js', 'css' ], path: 'brotli', plugins: [] }
Pattern: **/*.{js,css}
[ '1-71de5e33d838376c8246.js',
  'app-7bd68e3807465232107c.js',
  'component---src-pages-404-js-7d98b28b8cf43092d95f.js',
  'component---src-pages-index-js-326820e4ca9a4b241c8d.js',
  'component---src-pages-page-2-js-d2c2652b55fd265a0f89.js',
  'styles-6be7257993bc60dbd36e.js',
  'styles.fc4fa5e094d218207796.css',
  'webpack-runtime-2eced3bedeb760f3af0e.js' ]
info Done building in 10.426 sec
```

3. It breaks when `js`, `css` and `html` option are passed.
Config:
```js
{
  resolve: "gatsby-plugin-brotli",
    options: {
      extensions: ["js", "css", "html"],
      path: "brotli"
  }
}
```
Plugin code with `console.log()`: Same as above.
Output on Terminal:
```bash
success Building static HTML for pages - 0.369 s — 4/4 27.19 pages/second
PluginOptions: { plugins: [],
  extensions: [ 'js', 'css', 'html' ],
  path: 'brotli' }
AllOptions: { extensions: [ 'js', 'css', 'html' ],
  path: 'brotli',
  plugins: [] }
Pattern: **/*.{js,css,html}
[ '1-71de5e33d838376c8246.js',
  '404.html',
  '404/index.html',
  'app-7bd68e3807465232107c.js',
  'component---src-pages-404-js-7d98b28b8cf43092d95f.js',
  'component---src-pages-index-js-326820e4ca9a4b241c8d.js',
  'component---src-pages-page-2-js-d2c2652b55fd265a0f89.js',
  'index.html',
  'page-2/index.html',
  'page-data/404.html',
  'styles-6be7257993bc60dbd36e.js',
  'styles.fc4fa5e094d218207796.css',
  'webpack-runtime-2eced3bedeb760f3af0e.js' ]

 ERROR 

Plugin gatsby-plugin-brotli returned an error EISDIR: illegal operation on a directory, read

  Error: EISDIR: illegal operation on a directory, read

error Command failed with exit code 1.
```
The reason it breaks for `html`  is because the plugin lists a file`page-data/404.html,` as seen in above output on terminal. Whereas running `gatsby build` outputs a directory with the name `page-data/404.html` rather than a file. The plugin assumes its a file and tries to read it.

**Comment by ovhemert** [comment]
I've published a new version v1.2.3 that resolves the issues.
Please check if your build is all fine now.

@gagan0723 : Thanks for the donation BTW  🎉

**Comment by gagan0723** [comment]
The new version resolves the issue. Thanks
